ldap_auth_provider.py

```diff
@@ -480,33 +480,16 @@ def _ldap_authenticated_search(self, server, password, filters):
         """
 
         try:
-            conn = yield threads.deferToThread(
-                ldap3.Connection,
-                server,
-                self.ldap_bind_dn,
-                self.ldap_bind_password,
-                raise_exceptions=True,
-            )
-            logger.debug(
-                "Established LDAP connection in search mode: %s",
-                conn
+            result, conn = yield self._ldap_simple_bind(
+                server=server,
+                bind_dn=self.ldap_bind_dn,
+                password=self.ldap_bind_password,
             )
 
-            if self.ldap_start_tls:
-                yield threads.deferToThread(conn.open)
-                yield threads.deferToThread(conn.start_tls)
-                logger.debug(
-                    "Upgraded LDAP connection in search mode through "
-                    "StartTLS: %s",
-                    conn
-                )
+            if not result:
+                if hasattr(conn, "unbind"):
+                    yield threads.deferToThread(conn.unbind)
```

Comment:
It looks like the `_ldap_simple_bind` code already calls `unbind` during a failure: https://github.com/matrix-org/matrix-synapse-ldap3/blob/5a9b52c4d03b898ca2eda7f789ae3684c1ed584c/ldap_auth_provider.py#L453